### PR TITLE
fix #141

### LINKED ISSUE #141: svg: gradients are invisible if symbols are appended in DOM with display:none
For the context see similar discussion http://stackoverflow.com/questions/27752494/why-cant-i-reference-an-svg-linear-gradient-defined-in-an-external-file-paint

TODO add in a tutorial for preparing svg symbols a hint about where to put gradients definition
https://bugzilla.mozilla.org/show_bug.cgi?id=353575
ok for chrome:
```
<svg>
 <symbol>
  <defs><linearGradient.../>...</defs>
  <path ...>
 </symbol>
</svg>
```

ok (and more efficient) for chrome AND FF:
```
<svg>
  <defs><linearGradient.../>...</defs>
 <symbol>
  <path ...>
 </symbol>
</svg>
```
meaning: define gradients (and other stuff) outside `symbol`
should be explained with shadow DOM (?)

### PR DESCRIPTION
**Pull request by tyanas**
quick workaround

### CODE REVIEW

**Comment by lahmatiy** [comment]
Is it ok that `svgStorageElement` is visible? I think not. We should hide it.

File: src/basis.js [hunk 1]
```diff
@@ -3827,6 +3827,14 @@
       init: function(url){
         this.url = url;
         this.baseURI = pathUtils.dirname(url) + '/';
+
+        if (!svgStorageElement)
+        {
+          svgStorageElement = document.createElement('span');
+          /** @cut */ svgStorageElement.setAttribute('title', 'svg symbol storage');
+          svgStorageElement.style.cssText = 'position: absolute !important';
```
Comment (lahmatiy) [inline src/basis.js]: No spaces needed `position:absolute!important`

**Comment by tyanas** [comment]
> Is it ok that svgStorageElement is visible? 
if a gradient definition is in hidden DOM node (<something style="display:none">) then the gradient is invisible.
(ff bug at least https://bugzilla.mozilla.org/show_bug.cgi?id=376027)

CSS rules (for svgStorageElement ) like
```
    visibility: hidden;
    width: 0;
    height: 0;
    overflow: hidden;
```
do not break gradients visibility.

I can add any of these rules. What is the goal? @lahmatiy

**Comment by lahmatiy** [comment]
@tyanas The goal that `svgStorageElement` should to be shown. Try yourself:

```js
var uiNode = require('basis.ui').Node;
new uiNode({
    container: document.body,
    template: '<div><br/><b:svg src="./test.svg" use="#test"/></div>'
});
```

Where `test.svg`:

```svg
<svg width="200" height="250" version="1.1" xmlns="http://www.w3.org/2000/svg">
  <rect id="test" x="10" y="10" width="30" height="30" fill="rgba(0,100,0,.5)" stroke-width="5"/>
</svg>
```

You'll see two rects instead of one.

**Comment by tyanas** [comment]
I'd say `svg` without `defs` or `symbol` is not the case for `b:svg`. I'd rather add a dev warning when such svg file is used as a source in the tag.

If one wants to reuse an image (in terms of copying and pasting Dom nodes), one should try svg symbol. Maybe b:svg should be renamed as b:symbol
@rdvornov

**Pull request merged**